### DISCUSSION TITLE
Can you bind next/previous window?

### DISCUSSION THREAD

**Discussion by mitsuhiko** [ghostty-org/ghostty#3193] [Q&A] [answered]
On macOS you can use ``cmd+`​`` and ``cmd+shift+`​`` to cycle between windows.  This does not seem to be exposed to keybinds however or I'm missing something.

Basically looking for this:

```
keybind = super+right=next_window
keybind = super+left=previous_window
```

Is this possible today?

**Comment by mitchellh** [comment]
Not available. I'm open to doing this but I'm curious if macOS has an API to let you do this. Once I (or someone else) finds that, I can convert this to an issue, I just want to make sure it's possible. :)

**Reply by mitsuhiko** [reply]
I'm not sure if macOS has an API for this.  I just looked at what iTerm2 does and they manually keep track of the terminal windows open: https://github.com/gnachman/iTerm2/blob/885a8a89146262de7f05a21ddf9bba935c35c7cb/sources/iTermController.m#L416-L446

**Reply by mitchellh** [reply]
I can't look at iTerm2's source since it's GPL. You're welcome to describe it to me (it sounds like you already did though).

**Reply by mitsuhiko** [reply]
I just asked ChatGPT if it has a better idea and it came up with this, which is similar in nature:

```objc
- (void)goToNextWindow {
    NSArray<NSWindow *> *windows = [NSApp windows];
    NSWindow *keyWindow = [NSApp keyWindow];

    if (!keyWindow || windows.count == 0) return;

    NSUInteger currentIndex = [windows indexOfObject:keyWindow];
    NSUInteger nextIndex = (currentIndex + 1) % windows.count;

    NSWindow *nextWindow = windows[nextIndex];
    [nextWindow makeKeyAndOrderFront:self];
}

- (void)goToLastWindow {
    NSArray<NSWindow *> *windows = [NSApp windows];
    NSWindow *keyWindow = [NSApp keyWindow];

    if (!keyWindow || windows.count == 0) return;

    NSUInteger currentIndex = [windows indexOfObject:keyWindow];
    NSUInteger lastIndex = (currentIndex == 0) ? windows.count - 1 : currentIndex - 1;

    NSWindow *lastWindow = windows[lastIndex];
    [lastWindow makeKeyAndOrderFront:self];
}
```

**Reply by cmaahs** [reply]
Keybindings to jump next/previous in `tabs` and `windows` is likely the feature I use the most in wezterm.  It isn't apples to apples, wezterm being written in rust, but it is an MIT license.  

Not pertinent to the discussion, though the name, logo, and icon for the project are fantastic.

**Reply by hlubek** [reply]
This would be very helpful to add, since I cycle through windows of terminals a lot (`cmd+<`, `cmd+>`) and default keybindings in mac OS for cycling windows is really bad for a default german keyboard layout (backtick which is a deadkey).

**Comment by mitchellh** [comment, marked as answer]
Finally accepted #8387
